Code/Framework/AzToolsFramework/Tests/EditorViewportOnlyDrawHelpersForSelectedEntityTests.cpp

```diff
@@ -0,0 +1,186 @@
+        bool m_displayEntityViewportEvent = false;
+        AZStd::unique_ptr<ViewportSettingsTestImpl> m_viewportSettings;
+        AZStd::unique_ptr<AzToolsFramework::EditorHelpers> m_editorHelpers;
+        AZStd::unique_ptr<::testing::NiceMock<MockEditorVisibleEntityDataCacheInterface>> m_entityVisibleEntityDataCacheMock;
+        AzFramework::CameraState m_cameraState;
+    };
+
+    TEST_F(EditorViewportOnlyDrawHelpersForSelectedEntityFixture, DisplayDebugDrawIfSelectedEntitiesOptionDisabledAndEntitySelected)
+    {
+        // Given the entity is selected and the option to only show helpers for selected entities is false
+        const AzToolsFramework::EntityIdList entityIds = { m_entityId };
+        AzToolsFramework::ToolsApplicationRequestBus::Broadcast(
+            &AzToolsFramework::ToolsApplicationRequests::SetSelectedEntities, entityIds);
+
+        m_viewportSettings->m_onlyShowForSelectedEntities = false;
+
+        // When the draw function is called
+        m_editorHelpers->DisplayHelpers(
+            AzFramework::ViewportInfo{ TestViewportId },
+            m_cameraState,
+            *m_debugDisplay,
+            [](AZ::EntityId)
+            {
+                return true;
+            });
+
+        // Then the helper should be drawn
+        EXPECT_TRUE(m_displayEntityViewportEvent);
+    }
+
+    TEST_F(EditorViewportOnlyDrawHelpersForSelectedEntityFixture, DisplayDebugDrawIfSelectedEntitiesOptionDisabledAndEntityNotSelected)
+    {
+        // Given the entity is not selected and the option to only show helpers for selected entities is false
+        m_viewportSettings->m_onlyShowForSelectedEntities = false;
+
+        // When the draw function is called
+        m_editorHelpers->DisplayHelpers(
+            AzFramework::ViewportInfo{ TestViewportId },
+            m_cameraState,
+            *m_debugDisplay,
+            [](AZ::EntityId)
+            {
+                return true;
+            });
+
+        // Then the helper should be drawn
+        EXPECT_TRUE(m_displayEntityViewportEvent);
+    }
+
+    TEST_F(EditorViewportOnlyDrawHelpersForSelectedEntityFixture, DoNotDisplayDebugDrawIfSelectedEntitiesOptionEnabledAndEntityNotSelected)
+    {
+        // Given the entity is not selected and the option to only show helpers for selected entities is true
+        m_viewportSettings->m_onlyShowForSelectedEntities = true;
+
+        // When the draw function is called
+        m_editorHelpers->DisplayHelpers(
+            AzFramework::ViewportInfo{ TestViewportId },
+            m_cameraState,
+            *m_debugDisplay,
+            [](AZ::EntityId)
+            {
+                return true;
+            });
+
+        // Then the helper should not be drawn
+        EXPECT_FALSE(m_displayEntityViewportEvent);
+    }
+
+    TEST_F(EditorViewportOnlyDrawHelpersForSelectedEntityFixture, DisplayDebugDrawIfSelectedEntitiesOptionEnabledAndEntityIsSelected)
```

Comment:
I think these tests are a fantastic start but it would be super valuable to have a test with multiple entities in view, and we check only the selected ones draw with the setting enabled, and then all entities draw when it's disabled. I think this would be a really valuable thing to verify. Overall this is really strong though, I just think we can make them even better 🙂  Awesome work!